### PR TITLE
Fix ossl_shim; SSL_get_servername() during (TLS 1.2) resumption handshakes

### PR DESCRIPTION
**Pull request by kaduk**
Should fix the boringssl test issues brought in by #6378

### CODE REVIEW

File: ssl/ssl_lib.c [hunk 1]
```diff
@@ -2618,8 +2618,11 @@ const char *SSL_get_servername(const SSL *s, const int type)
      * peer send" and "what was actually negotiated"; we should have
      * a clear distinction amongst those three.
      */
-    if (SSL_in_init(s))
+    if (SSL_in_init(s)) {
+        if (s->hit)
+            return s->session->ext.hostname;
```
Comment (tmshort) [inline ssl/ssl_lib.c]: Do you need to check `s->session != NULL` as is done below?

Comment (kaduk) [reply ssl/ssl_lib.c]: I believe that it's impossible to have a resumed SSL with no associated SSL_SESSION.
(But I'm happy to add a check if we don't want to rely on that as an invariant.)

Comment (tmshort) [reply ssl/ssl_lib.c]: True, you have the check on `s->hit`...

**Comment by levitte** [comment]
Hmmm, style wise, should ossl_shim.cc follow our style guide, or nah?

**Comment by kaduk** [comment]
> Hmmm, style wise, should ossl_shim.cc follow our style guide, or nah?

Well, first off, we don't have a C++ style guide :)

But this is (AFAIU) basically a clone of bssl_shim with some bits ripped out and a few tweaks, so leaving it as the boringssl style seems like the best choice to me.

**Comment by levitte** [comment]
Good enough for me

**Comment by levitte** [comment]
Feel free to edit the message 🙂

**Comment by kaduk** [comment]
@levitte just to be clear: does approval still hold with edited commit message?

**Comment by kaduk** [comment]
Pushed to master; closing.
Thanks for the review!

**Pull request closed without merging**